cpp/src/lists/copying/scatter_helper.cu

```diff
@@ -254,9 +254,10 @@ struct list_child_constructor {
         auto child_strings_column = lists_column.child();
         auto strings_offset       = lists_offsets_ptr[row_index] + intra_index;
 
-        return child_strings_column.is_null(strings_offset)
-                 ? string_view{nullptr, 0}
-                 : child_strings_column.template element<string_view>(strings_offset);
+        if (child_strings_column.is_null(strings_offset)) { return string_view{nullptr, 0}; }
+        auto const d_str = child_strings_column.template element<string_view>(strings_offset);
+        // ensure a string from an all-empty column is not mapped to the null placeholder
+        return d_str.empty() ? string_view{} : d_str;
```

Comment:
Why the difference in 'empty' string returns. In one case we pass `string_view{nullptr,0}` and the other `string_view{}`?

Reply:
Could we have `make_empty_view` and `make_null_view` functions to show the intent differences?

Reply:
I suppose but it would only apply for this factory function. The placeholder is passed in so theoretically someone could use a different `string_view` to identify a null entry for use with this factory.

Reply:
I think this makes sense. I've added named variables to hopefully make the intent more clear.

Reply:
+1, I like this idea. It really improves clarity.